pyomo/contrib/parmest/parmest.py

```diff
@@ -585,19 +496,22 @@ def _Q_opt(self, ThetaVals=None, solver="ef_ipopt",
                 (7-5-16) in "Nonlinear Parameter Estimation", Y. Bard, 1974.
                 
                 This formula is also applicable if the objective is scaled by a constant;
-                the constant cancels out. (PySP scaled by 1/n because it computes an
+                the constant cancels out. (was scaled by 1/n because it computes an
                 expected value.)
                 '''
                 cov = 2 * sse / (n - l) * inv_red_hes
                 cov = pd.DataFrame(cov, index=thetavals.keys(), columns=thetavals.keys())
             
             if len(return_values) > 0:
                 var_values = []
-                for exp_i in self.ef_instance.component_objects(Block, descend_into=False):
+                # assumes we solved using mpi-sppy
+                for exp_i in ef.component_objects(Block, descend_into=False):
                     vals = {}
                     for var in return_values:
                         exp_i_var = exp_i.find_component(str(var))
-                        temp = [pyo.value(_) for _ in exp_i_var.itervalues()]
+                        if exp_i_var is None:  # we might have a block such as _mpisppy_data
+                            continue
+                        temp = [pyo.value(_) for _ in exp_i_var.values()]
                         if len(temp) == 1:
                             vals[var] = temp[0]
                         else:
```

Comment:
The current code adds additional rows of NaNs when exp_i_var is None. I suggest adding the following lines at 519.
```
if len(vals) > 0:
    var_values.append(vals) 
```

Reply:
Thanks! This solves the mystery.  We could do the fix you suggest, or maybe do something to avoid looking at the
extra blocks; either way, this should be the last piece of the puzzle.